src/base/bittorrent/session.cpp

```diff
@@ -370,12 +370,12 @@ Session::Session(QObject *parent)
     , m_isIPFilteringEnabled(BITTORRENT_SESSION_KEY("IPFilteringEnabled"), false)
     , m_isTrackerFilteringEnabled(BITTORRENT_SESSION_KEY("TrackerFilteringEnabled"), false)
     , m_IPFilterFile(BITTORRENT_SESSION_KEY("IPFilter"))
-    , m_announceToAllTrackers(BITTORRENT_SESSION_KEY("AnnounceToAllTrackers"), false)
+    , m_announceToAllTrackers(BITTORRENT_SESSION_KEY("AnnounceToAllTrackers"), true)
     , m_announceToAllTiers(BITTORRENT_SESSION_KEY("AnnounceToAllTiers"), true)
     , m_asyncIOThreads(BITTORRENT_SESSION_KEY("AsyncIOThreadsCount"), 4)
     , m_filePoolSize(BITTORRENT_SESSION_KEY("FilePoolSize"), 40)
-    , m_checkingMemUsage(BITTORRENT_SESSION_KEY("CheckingMemUsageSize"), 16)
-    , m_diskCacheSize(BITTORRENT_SESSION_KEY("DiskCacheSize"), 64)
+    , m_checkingMemUsage(BITTORRENT_SESSION_KEY("CheckingMemUsageSize"), 64)
+    , m_diskCacheSize(BITTORRENT_SESSION_KEY("DiskCacheSize"), -1)
```

Comment:
This will take up too much memory than the user expectation and they will complain.

Reply:
Which one, specifically? If the DiskCacheSize is on auto, the OS will manage it as needed as cached memory and free it for application that needed as far as I know, so there should be  no problem. If user complains are something like [this](https://www.linuxatemyram.com/), it is not an issue.

 As for CheckingMemUsageSize, on systems with very low available memory like the early/less powerful RPis (256 - 512 MiB), I can see this being the straw that breaks the camel's back. But isn't it a one-time cost, since qBittorrent only checks one torrent at a time? Also I would argue that early model Pis fall into the "extreme end" of lower performance machines, I would not say they are part of the common case, but then again, I don't have data to back this up, it is just an opinion. What about simply raising it to 32 MiB instead, as a compromise?

Reply:
>If the DiskCacheSize is on auto, the OS will manage it as needed as cached memory and free it for application that needed as far as I know, so there should be no problem.

That is not quite right, see this:
https://www.libtorrent.org/single-page-ref.html#cache_size
https://github.com/arvidn/libtorrent/blob/d33b0506a0f4cd61860fd54864267c82d3e171e3/src/disk_buffer_pool.cpp#L287

>If user complains are something like this, it is not an issue.

LOL

>What about simply raising it to 32 MiB instead, as a compromise?

OK for me.

Reply:
> > If the DiskCacheSize is on auto, the OS will manage it as needed as cached memory and free it for application that needed as far as I know, so there should be no problem.
> 
> That is not quite right, see this:
> https://www.libtorrent.org/single-page-ref.html#cache_size
> https://github.com/arvidn/libtorrent/blob/d33b0506a0f4cd61860fd54864267c82d3e171e3/src/disk_buffer_pool.cpp#L287
> > If user complains are something like this, it is not an issue.
> 
> LOL

What's funny? Did you see the link? I am simply saying that users complaining about memory that is "used", but actually available for other programs is not a problem. However, as you pointed out, it seems I was wrong about the nature of libtorrent's memory allocation for the disk cache. Still, having looked at the source code you linked, the actual allocated values in auto mode seem quite reasonable:

```C++
// we take a 40th of everything exceeding 4 GiB
// a 30th of everything exceeding 1 GiB
// and a 10th of everything below a GiB
```

Some examples:

- 512 MiB RAM: 512/10 = ~51,2 MiB allocated for the disk cache (actually a bit smaller than the current default, which with this little RAM is probably a good thing)
- 1 GiB RAM: 1024/10 = ~100 MiB (only slightly bigger)
- 2 GiB RAM: 2048/30 = ~68 MiB (about on par)
- 4 GiB RAM: 4096/30 = ~136 MiB (bigger, not unreasonable)
- 8 GIB RAM: 8192/40 = ~200 MiB (bigger, not unreasonable)
- 16 GiB RAM: 16384/40 = ~400 MiB (bigger, not unreasonable)
...

For 2 GiB and less, the values are actually similar to the current default. For higher amounts, the values are bigger, but still a rather small percentage vs. total system memory, while providing a nice performance boost. Additionally, it is 1000x better than the snake-oil amounts I see users allocate manually. I have seen many issues/comments here about users trying to optimize read/write performance, by allocating silly values like 1 GiB, 2 GiB, 3 GiB or more to the cache which at best is just wasteful and at worst is actually detrimental to performance.

Reply:
>What's funny? Did you see the link? I am simply saying that users complaining about memory that is "used", but actually available for other programs is not a problem. 

I hope you're not really offended... I was referring to the top picture in the link.

>I am simply saying that users complaining about memory that is "used", but actually available for other programs is not a problem. 

But seriously I don't see many people complain that on our issue tracker.

>Some examples:

Seems reasonable. Also if any low-end system run into problem, I think I'll simply redirect the complain to libtorrent to revise their formula.

Reply:
> > What's funny? Did you see the link? I am simply saying that users complaining about memory that is "used", but actually available for other programs is not a problem.

> I hope you're not really offended... I was referring to the top picture in the link.

I didn't take anything personally, I realized that if you hadn't seen the link, the whole sentence sound like "it is not a problem if users complain about qbt using more memory" which is a bit silly without context.

> But seriously I don't see many people complain that on our issue tracker.

Me neither so far, let's hope it stays that way.

> Seems reasonable. Also if any low-end system run into problem, I think I'll simply redirect the complain to libtorrent to revise their formula.

Ok. The current formula seems fair for low-end systems (though there is that outlier "jump" @ 1 GiB). I'd actually be more concerned with the fact that there is no additional branch in the formula beyond 4 GiB. Users with higher end machines with 32 GiB, 64 GiB or higher will get caches of 800  MiB, 1.6 GiB and beyond, which is almost certainly wasteful. Though with that much total system memory, that is very much a first-world problem, and like you said, a problem to be solved on libtorrent's side anyway.